ext/bz2/bz2_filter.c

```diff
@@ -367,6 +367,10 @@ static php_stream_filter *php_bz2_filter_create(const char *filtername, zval *fi
 					zend_long blocks = zval_get_long(tmpzval);
 					if (blocks < 1 || blocks > 9) {
 						php_error_docref(NULL, E_WARNING, "Invalid parameter given for number of blocks to allocate (" ZEND_LONG_FMT ")", blocks);
+						pefree(data->strm.next_in, persistent);
```

Comment:
nit: only if you want, on master only maybe, making a goto label to cover the 3 cases.